CLDR-15223 feedback for adding users

srl295 opened the pull request:
- add API endpoint for locale normalizer
- add LocaleNormalizer.getMessageArray()
- live feedback on add user panel

CLDR-15223

- [ ] This PR completes the ticket.

image below from adding `und doi ff es FF` as locales  for an Adlam user

![image](https://user-images.githubusercontent.com/855219/169412042-c2ad7382-d10a-45bf-bb35-fba82cdd7b8b.png)

the field gets updated to just 'ff'

macchiati commented:
"Locale list was updated due to:"

That is the *opposite* of what the message wants to be.

What we want to say is:

The following locales were *not* added due to problems:
- **und** — the locale "und" does not match an individual language
- **doi (Dogri)** — is outside of your organization's coverage
- FF — does not have the right casing for a language code
- **es (Spanish)** — is outside of your organization's coverage
- $$ — is not a valid language code

(I added the $$ as a weird case)

macchiati reviewed (changes requested):
(see comment)

macchiati commented:
PS, I can see the rationale for not accepting FF. While it is a valid language code, it is not in canonical form, and might be from the user's typing in a region code by mistake.

srl295 commented:
> "Locale list was updated due to:"
> 
> 
> 
> That is the *opposite* of what the message wants to be.

Good feedback. Quick notes:

- "updated" means what you typed was normalized. It does this when you exit the field. 
- the normalization and messages are pre existing. But can definitely work on the wording. 
- I'll add a test for the normalization logic if there isn't one.

DavidLRowe commented:
Does this check (for example) a user entering "kmr" (ISO 639-3 code for northern Kurdish), when "ku" is what is needed?

macchiati commented:
I like that. Have a low-level common method to check a locale string, with
options. Returns locale string (or und), and fills enum set

Enum set indicates whether canonicalized and how, whether disallowed and
how. Each enum value has a message.

On Fri, May 20, 2022, 07:41 Steven R. Loomis ***@***.***>
wrote:

> El El vie, may. 20, 2022 a la(s) 8:42 a.m., DavidLRowe <
> ***@***.***> escribió:
>
> > Does this check (for example) a user entering "kmr" (ISO 639-3 code for
> > northern Kurdish), when "ku" is what is needed?
> >
>
>
> I’m not sure. If kmr is not a CLDR locals it won’t allow it. But I don’t
> know if it will map it.
>
> Perhaps we could merge this with improvements to the outer ui and have a
> separate pr for improving the messages and checking overlong cases.
>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/2033#issuecomment-1132984159>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMHQ772YPAM5QRTMAJLVK6QBTANCNFSM5WNSRHCQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

srl295 commented:
> I like that. Have a low-level common method to check a locale string, with options. Returns locale string (or und), and fills enum set Enum set indicates whether canonicalized and how, whether disallowed and how. Each enum value has a message.
> […](#)

it's LocaleNormalizer and it was added in #1290 a year ago.

srl295 commented:
> Does this check (for example) a user entering "kmr" (ISO 639-3 code for northern Kurdish), when "ku" is what is needed?

`kmr` (and `eng` for that matter) return just 'unknown'.

srl295 commented:
OK current update. 

* First, removed 'example' and made it a placeholder:
![image](https://user-images.githubusercontent.com/855219/169581299-513c0347-59a5-4ab1-8d08-0aae41ff8f3a.png)
(shows as grayed out `en de de_CH…` if you delete all text)

* Next, changed LocaleNormalizer to return enums.    Localizable in cldrText.js.  Also, use a display name wherever we have it.

![image](https://user-images.githubusercontent.com/855219/169581514-281cffde-4c4d-47e3-ac22-46f30e315d0e.png)

If the UI itself looks OK, let's iterate on the normalizer's behavior separately, because it's already called from many other spots.

DavidLRowe commented:
That's probably reasonable. Good.

macchiati commented:
LGTM

On Fri, May 20, 2022, 14:16 DavidLRowe ***@***.***> wrote:

> That's probably reasonable. Good.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/2033#issuecomment-1133385150>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMGJDSOWIYCU6UQJDFLVK76LJANCNFSM5WNSRHCQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
